metadata-ingestion/src/datahub/ingestion/source/powerbi/proxy.py

```diff
@@ -0,0 +1,911 @@
+            return hash(self.__members())
+
+    def __init__(self, config: PowerBiAPIConfig) -> None:
+        self.__config: PowerBiAPIConfig = config
+        self.__access_token: str = ""
+        # Power-Bi Auth (Service Principal Auth)
+        self.__msal_client = msal.ConfidentialClientApplication(
+            self.__config.client_id,
+            client_credential=self.__config.client_secret,
+            authority=PowerBiAPI.AUTHORITY + self.__config.tenant_id,
+        )
+
+        # Test connection by generating a access token
+        LOGGER.info("Trying to connect to {}".format(self.__get_authority_url()))
+        self.get_access_token()
+        LOGGER.info("Able to connect to {}".format(self.__get_authority_url()))
+
+    def __get_authority_url(self):
+        return "{}{}".format(PowerBiAPI.AUTHORITY, self.__config.tenant_id)
+
+    def __get_users(self, workspace_id: str, entity: str, _id: str) -> List[User]:
+        """
+        Get user for the given PowerBi entity
+        """
+        users: List[PowerBiAPI.User] = []
+        if self.__config.extract_ownership is False:
+            LOGGER.info(
+                "ExtractOwnership capabilities is disabled from configuration and hence returning empty users list"
+            )
+            return users
+
+        user_list_endpoint: str = PowerBiAPI.API_ENDPOINTS[Constant.ENTITY_USER_LIST]
+        # Replace place holders
+        user_list_endpoint = user_list_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=PowerBiAPI.ADMIN_BASE_URL,
+            ENTITY=entity,
+            ENTITY_ID=_id,
+        )
+        # Hit PowerBi
+        LOGGER.info(f"Request to URL={user_list_endpoint}")
+        response = requests.get(
+            user_list_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+
+        # Check if we got response from PowerBi
+        if response.status_code != 200:
+            LOGGER.warning(
+                "Failed to fetch user list from power-bi. http_status=%s. message=%s",
+                response.status_code,
+                response.text,
+            )
+
+            LOGGER.info(f"{Constant.WorkspaceId}={workspace_id}")
+            LOGGER.info(f"{Constant.ENTITY}={entity}")
+            LOGGER.info(f"{Constant.ID}={_id}")
+            raise ConnectionError("Failed to fetch the user list from the power-bi")
+
+        users_dict: List[Any] = response.json()[Constant.VALUE]
+
+        # Iterate through response and create a list of PowerBiAPI.Dashboard
+        users = [
+            PowerBiAPI.User(
+                id=instance.get("identifier"),
+                displayName=instance.get("displayName"),
+                emailAddress=instance.get("emailAddress"),
+                graphId=instance.get("graphId"),
+                principalType=instance.get("principalType"),
+            )
+            for instance in users_dict
+        ]
+
+        return users
+
+    def __get_report(
```

Comment:
do we need 2 underscores here? in these method names. do we do this in other places?

Reply:
Updated it to __get_report_